internal/controller/suite_test.go

```diff
@@ -64,7 +66,11 @@ var _ = BeforeSuite(func() {
 
 	By("bootstrapping test environment")
 	testEnv = &envtest.Environment{
-		CRDDirectoryPaths:     []string{filepath.Join("..", "..", "config", "crd", "bases")},
+		CRDDirectoryPaths: []string{
+			filepath.Join("..", "..", "config", "crd", "bases"),
+			filepath.Join("..", "..", "test", "configs", "service-monitor.yaml"),
```

Comment:
i think i misled you here because i thought that we already had those resources locally :/ 

a better test indeed might be in e2e - we could simply, in the first step, check against the prometheus to see if it reports the up metric for the component.

after deleting the SM we can just check that its gone. My apologies there I didnt check thoroughly enough the structure

Reply:
I think it is fine to add the test here, but I wonder if we need to include CRD this way 🤔 

@coleenquadros, I think the Environment struct has a CRDs field that can take in a Go CustomResourceDefinition struct. I wonder if we can import that struct from Prometheus Operator somehow? That way we don't need to maintain the CRD in-tree

Reply:
maintaining the CRD in-tree is what I have the issue with. Ideally we would keep the integration tests as light as possible and enable running locally with out any network for example, as they are now.

Why I am suggesting adding this to e2e now instead is because we already take care of installing Prom operator there from my understanding and we can push much of the slower e2e work to ci. Thoughts?

Reply:
Yup that works too, essentially skip testing this in integration.